### PR TITLE
Docs: Webpack changes for addons

### PR DESCRIPTION
**Pull request by jonniebigodes**
Closes #

## What I did

With this pull request the Webpack documentation is updated to feature the newly created Webpack addons

What was done:
- Updated the documentation
- Created snippets
- Fixed some headings and callouts that were incorrectly rendered

@valentinpalkovic I've assigned you to this so that you can take a look at this and follow up with me on it so that we can adjust as needed if required. Let me know and we'll take it from there. Thanks in advance.

### Testing

1. Follow the steps in the [contributing instructions](https://storybook.js.org/docs/react/contribute/new-snippets#preview-your-work) for this branch, `docs_webpack_changes_for_addons`.
2. Open the relevant documentation and verify the feature changes.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: docs/api/main-config-babel-default.md [hunk 1]
```diff
@@ -26,7 +26,9 @@ To adjust your Storybook's Babel setup directly—not via an addon—use [`babel
 
 ## `Babel.Config`
 
-See [Babel docs](https://babeljs.io/docs/options).
+The options provided by [Babel](https://babeljs.io/) are only applicable if you've enabled the [`@storybook/addon-webpack5-compiler-babel`](https://storybook.js.org/addons/@storybook/addon-webpack5-compiler-babel) addon.
```
Comment (kylegach) [inline docs/api/main-config-babel-default.md:29]: Why the change in link URL? The previous one was more helpful, imo.

**Review by kylegach — APPROVED** [review]
LGTM!

File: docs/api/main-config-babel-default.md (hunk 1 above)
Comment (jonniebigodes) [reply docs/api/main-config-babel-default.md]: This was accidentally introduced, I forgot to update the link accordingly, but it should be all fixed now

File: docs/builders/webpack.md [hunk 2]
```diff
@@ -107,9 +107,57 @@ Additionally, if you're generating a [static build](../api/cli-options.md#build)
 
 <!-- prettier-ignore-end -->
 
-## What about Webpack 4 support?
+## Compiler support
 
-Support for Webpack 4 has been removed and is no longer being maintained. If you're upgrading your Storybook, it will automatically use Webpack 5 and attempt to migrate your configuration. However, if you're working with a custom Webpack configuration, you may need to update it to work with Webpack 5. The migration process is necessary to ensure that your project runs smoothly with the latest version of Storybook. You can follow the instructions provided on the Webpack [website](https://webpack.js.org/migrate/5/) to update your configuration.
+Out of the box, Storybook provides a compiler-agnostic approach to bundling your project, allowing you to use the same tooling you use to bundle your application (e.g., [Babel](https://babeljs.io/), [SWC](https://swc.rs/)), enabling efficient and performant component development within the ecosystem of Webpack 5-based projects.
```
Comment (chantastic) [inline docs/builders/webpack.md:112]: I'd break this sentence up, for readability.
Not a deal breaker but the following may read cleaner:

```md
Storybook takes a compiler-agnostic approach to bundling. This allows you to bring your own application bundler (e.g., [Babel](https://babeljs.io/), [SWC](https://swc.rs/)) and ensures compatibility within the vast ecosystem of Webpack 5-based projects.
```

File: docs/builders/webpack.md [hunk 3]
```diff
@@ -107,9 +107,57 @@ Additionally, if you're generating a [static build](../api/cli-options.md#build)
...
+
+### SWC
+
+If you're working with a project built using [SWC](https://swc.rs/) for faster builds and improved performance, you can use the [`@storybook/addon-webpack5-compiler-swc`](https://storybook.js.org/addons/@storybook/addon-webpack5-compiler-swc) addon, allowing you the same performance benefits and compatibility with your project. Run the following command to set up the addon automatically:
```
Comment (chantastic) [inline docs/builders/webpack.md:116]: May be improved by breaking up sentances.

```
If your project is built using [SWC](https://swc.rs/), use the [`@storybook/addon-webpack5-compiler-swc`](https://storybook.js.org/addons/@storybook/addon-webpack5-compiler-swc) addon. This addon increases ecosystem compatibility, with with Webpack 5 projects, while maintaining high performance. Run the following command to set up the addon automatically:
```

File: docs/builders/webpack.md [hunk 4]
```diff
@@ -107,9 +107,57 @@ Additionally, if you're generating a [static build](../api/cli-options.md#build)
...
+
+<!-- prettier-ignore-end -->
+
+<Callout variant="info">
+
+Additional options can be provided to customize the SWC configuration. See the [SWC API documentation](../api/main-config-swc.md) for more information.
+
+</Callout>
+
+When enabled, the addon will adjust the Webpack configuration to use the [`swc-loader`](https://swc.rs/docs/usage/swc-loader) as the default loader for JavaScript and TypeScript files and will detect and use your project's SWC configuration.
```
Comment (chantastic) [inline docs/builders/webpack.md:136]: split into multiple sentences.

```
When enabled, this addon adjusts the Webpack configuration to use the [`swc-loader`](https://swc.rs/docs/usage/swc-loader) for JavaScript and TypeScript files. Additionally, it will detect and use your project's SWC configuration.
```

**Review by chantastic — APPROVED** [review]
Approved, with a few suggestions on improving copy.

None of it is a show-stopper for me. But, given our wide range of readers, I felt that shorter, less conditional, sentences may improve readability.

**Review by valentinpalkovic — APPROVED** [review]
